@xen-orchestra/lite/src/components/form/FormCheckbox.vue

```diff
@@ -0,0 +1,189 @@
+<template>
+  <component
+    :is="hasLabel ? 'span' : 'label'"
+    :class="`form-${type}`"
+    v-bind="wrapperAttrs"
+  >
+    <input
+      v-model="value"
+      :disabled="isLabelDisabled || disabled"
+      :type="type === 'radio' ? 'radio' : 'checkbox'"
+      class="input"
+      v-bind="$attrs"
+    />
+    <span class="fake-checkbox">
+      <UiIcon :fixed-width="false" :icon="icon" class="icon" />
+    </span>
+  </component>
+</template>
+
+<script lang="ts">
+export default {
+  name: "FormCheckbox",
+  inheritAttrs: false,
+};
+</script>
+
+<script lang="ts" setup>
+import {
+  type HTMLAttributes,
+  type InputHTMLAttributes,
+  computed,
+  inject,
+  ref,
+} from "vue";
+import { faCheck, faCircle } from "@fortawesome/free-solid-svg-icons";
+import { useVModel } from "@vueuse/core";
+import UiIcon from "@/components/ui/UiIcon.vue";
+
+interface Props extends Omit<InputHTMLAttributes, ""> {
+  modelValue?: unknown;
+  disabled?: boolean;
+  wrapperAttrs?: HTMLAttributes;
+}
+
+const props = defineProps<Props>();
+
+const emit = defineEmits<{
+  (event: "update:modelValue", value: boolean): void;
+}>();
+
+const value = useVModel(props, "modelValue", emit);
+const type = inject<"checkbox" | "radio" | "toggle">("inputType", "checkbox");
+const hasLabel = inject("hasLabel", false);
+const isLabelDisabled = inject("isLabelDisabled", ref(false));
```

Comment:
It seems I misread your question, which was about the `isLabelDisabled`.

This way we can specify the `disabled` prop only on label instead of both inputs.

```html
<FormLabel :disabled="conditionToBeDisabled">
  <FormCheckbox />
  Accept something
</FormLabel>
```

Instead of:

```html
<FormLabel :disabled="conditionToBeDisabled">
  <FormCheckbox :disabled="conditionToBeDisabled">
  Accept something
</FormLabel>
```